### ISSUE TITLE
Femtocell detection & protection

### ISSUE THREAD

**Issue by SecUpwN** [SecUpwN/Android-IMSI-Catcher-Detector#6]
[Femtocells](https://en.wikipedia.org/wiki/Femtocell) are home routers that use broadband connections to improve mobile coverage, allowing calls to be made indoors more easily. While a femtocell all by itself is not necessarily harmful, they can be hacked ([as done in Great Britain by THC in 2010](https://wiki.thc.org/vodafone)) and be used to call anyone using the victims caller ID, read all SMS, MMS, listen to the voice mailbox and even intercept and record all phone calls made by the person who connected to such femtocell.

And while crawling the interwebz, I found a very interesting project by @iSECpartners: The [FemtoCatcher Android App for Verizon Android Smartphones](https://play.google.com/store/apps/details?id=com.isecpartners.femtocatcher). And the best thing of it all: It's [OpenSource on GitHub](https://github.com/iSECPartners/femtocatcher)! Their App has been presented at Black Hat and Defcon 21: "[Traffic Interception and Remote Mobile Phone Cloning with a Compromised CDMA Femtocell](https://www.youtube.com/watch?v=gfcq8clu1RI)", a short summary with presentation slides can be found on [their website](https://www.isecpartners.com/blog/2013/august/femtocell-presentation-slides-videos-and-app.aspx). Some features of it:

* FemtoCatcher uses the network ID information available through Android API calls to determine if the phone is connected to a Femtocell.
* They did not test how easy it would be for an attacker to change this information to fool the app, but certainly don’t rule out the possibility.
* Some Verizon Android phones display an icon in the status bar and/or display an ERI banner of “Network Extender” when connected to a femtocell. The strategy used by FemtoCatcher to detect the presence of a femtocell is based on the same techniques used by these indicators in Verizon ROMs.
* FemtoCatcher will not automatically take your phone out of airplane mode when you move away from a femtocell. You will be without service until you manually re-enable your connectivity. If FemtoCatcher is running and you are in range of a femtocell when you disable airplane mode, FemtoCatcher will quickly put your phone back in airplane mode.

I remember that when I discovered their project a while back, I even wrote an E-Mail to @tomrittervg of @iSECpartners introducing our project. He already saw our project and was very excited about it, telling me that they built FemtoCatcher with the explicit goal of having someone extend upon it. Sounds awesome, huh? ;-) At the time of my E-Mail he was travelling a lot, but he said that although they probably can not contribute much to our project developer wise, he'd send out a general link and tell folks if it looks interesting to get in touch for contributing. Very sympathic guy. Before your head starts smoking: What's your opinion, @xLaMbChOpSx? Could you add their femtocell detection and protection mechanism, maybe enhancing it to not only work with Verizon smartphones? Let me know if you need anything.

**Comment by TPS** [comment]
@SecUpwN Is it worth updating https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/wiki/Similar-Projects (Software Projects section) with the results of https://www.google.com/search?q=gsm+femtocell+detect ?

The specific results I'd think are relevant are:

https://play.google.com/store/apps/details?id=com.neutona.mobile.smallcelldetector

https://play.google.com/store/apps/details?id=com.bluejaysystools.mobilecrush
https://play.google.com/store/apps/details?id=bluejaysoftware.b4a.utility.femtowidgetlite
https://play.google.com/store/apps/details?id=bluejaysoftware.b4a.utility.femtowidget
https://play.google.com/store/apps/details?id=bluejaysoftware.b4a.utility.femtowidgetpro

These are all officially commercial, but do seem to have worked with GSM femtocells. (The last 4 are from the same developer.) Would it even be worth to contact them?

(This comment is re: your note to me from #457 about helping this issue.… ☺)

**Comment by He3556** [comment]
i was checking your links and found this. A RF Analyser APP that works with a cheap RTL-SDR over USB. This could be a serios solution for some of our hardest problems. 

https://play.google.com/store/apps/details?id=com.mantz_it.rfanalyzer

**Comment by E3V3A** [comment]
So for GSM/UMTS femtocells, we have to map each countries operators LAC's to what they reserve for their femtocells, at least if we're to believe this:

![ray_3-tpwu4_sgmt-yyro_yfhgme55atogjuz3mnvmt6ejo3zks6lp4qbtyjogmiwck h900](https://cloud.githubusercontent.com/assets/194392/8635448/f757ad0a-282b-11e5-93a6-f206d2dfe3e5.jpg)

**Comment by E3V3A** [comment]
Also pay attention to another very related [Google bug]9https://code.google.com/p/android/issues/detail?id=32593&q=bkphorn&colspec=ID%20Type%20Status%20Owner%20Summary%20Stars) that was closed without explanation and shown to be still valid... 

This may also explain some weird behavior I've seen on my UMTS network..

**Comment by SecUpwN** [comment]
@TPS, thank you very much for your valuable contributions here. I will get in touch with the developers.

**Comment by TPS** [comment]
@SecUpwN Since most of everything else I've attempted to contribute has seemed to be extraneous, I hesitated to even post the above when I thought this up last week (seems so obvious to attempt in retrospect, doesn't it, but Google *does* seem to continue to improve some parts of what they do to find stuff [I **do** understand the sentiment is unpopular here ;]), but I scoured the site docs & couldn't find any references to these.

I'm just glad I'm helping, for once! ☺

**Comment by TPS** [comment]
@He3556 You should know that https://play.google.com/store/apps/details?id=com.mantz_it.rfanalyzer is GPL ≥2, in spite of being paid on Play, with source @ https://github.com/demantz/RFAnalyzer ☺

**Comment by DJaeger** [comment]
@He3556: I don't think this is solved!?

**Comment by He3556** [comment]
A Femtocell is also a special case and depends on the different providers. 
So we had a long way getting a list or something, to be able to detect them. 
A nice to have - but in general a Femtocell must not be a security problem. 
If there are other projects doing this - we could implement a working solution at a later time.

**Comment by agilob** [comment]
> So we had a long way getting a list or something, to be able to detect them. 

>we could implement a working solution at a later time.

And... in both cases it shouldn't be closed.

**Comment by DJaeger** [comment]
I have to second @agilob
We have choosen to take care of Femtocell and also invested to much time to throw it all away!

**Comment by SecUpwN** [comment]
>We have choosen to take care of Femtocell and also invested to much time to throw it all away!

Indeed, I agree on this. But I am wondering how we're going to solve this Issue of we chose to make AIMSICD prompt that it only supports GSM networks as discussed in #489 while at the same time we only have code for CDMA FemtoCells in our app since the ones for GSM are missing - sounds like a loophole to me.

**Comment by javigc3** [comment]
Hi, @SecUpwN id like to contact you im very interested in financing your project please let me know if you are free and interested about it still thanks. i couldnt find a direct way to contact you my email is javigc3@gmail.com thanks
